airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -127,17 +206,34 @@ def state_checkpoint_interval(self) -> int:
 
     # Setting the default cursor field for all streams
     cursor_field = "updated_at"
+    deleted_cursor_field = "deleted_at"
+
+    @property
+    def deleted_events_instance(self) -> ShopifyDeletedEventsStream:
+        return ShopifyDeletedEventsStream(self.config, self.data_field, self.events_name)
 
     @property
     def default_state_comparison_value(self) -> Union[int, str]:
         # certain streams are using `id` field as `cursor_field`, which requires to use `int` type,
         # but many other use `str` values for this, we determine what to use based on `cursor_field` value
         return 0 if self.cursor_field == "id" else ""
 
+    @property
+    def default_deleted_state_comparison_value(self) -> Union[int, str]:
```

Comment:
it's a bit confusing this is hardcoded to be an empty string. Shouldn't it be `start_date` ?

Reply:
We don't provide the `default`  start date value within code now, we made this optional for all new users and all existing users, thus There is no guarantee we fall back correctly once we face a `missing` start date for existing connections. Therefore the empty string was used as a base comparison value, which makes life easier). The main rule works here is - that we made the start date optional, Let's not consider that value as a benchmark, if it's optional, an empty string works just fine here as it was before with the regular cursor_field.